src/main/java/org/terasology/climateConditions/TemperatureGeneratorSystem.java

```diff
@@ -42,7 +46,37 @@ public void preBegin() {
                 new ConditionModifier() {
                     @Override
                     public float getCondition(float value, float x, float y, float z) {
-                        return getValue(value, x, y, z);
+                        Vector3i position = new Vector3i(x, y, z);
+                        if (y <= environmentSystem.seaLevel) {
+                            return environmentSystem.temperatureBase;
+                        } else {
+                            // The higher above sea level - the colder (changes ~.03 degrees C per 1 meter change)
+                            // Temperature decreased by the height above sea level times .00006 as an exaggerated approximation
+                            float modifier = environmentSystem.temperatureBase * (1 + value * .07f) - (position.y - environmentSystem.seaLevel) * .00006f + .07f;
```

Comment:
Just to keep things realistic perhaps the [Lapse Rate](https://en.wikipedia.org/wiki/Lapse_rate) values can be used values like 5 degrees/km and 9.8 degrees/km can be used and are suggested there for different humidity conditions. I think that can simplify the logic.
(I am really sleepy rn, so it is quite possible that I may have misunderstood something here)

Reply:
Do you treat the numeric tempereature values as degrees Celsius in some way? What does 0 and 1 mean here then?

Reply:
Temperature is Celsius / 100. So 0 is 0 degrees C, while 1 is 100 degrees C.